File: archive/unique-morse-code-words.js
const uniqueMorseRepresentations = function(words) {
  let alphabet = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
  let set = new Set()

  words.forEach(word => {
    let morze = ''
    word.split('').forEach(w => {
      morze+=alphabet[w.charCodeAt(0) - 97]
    })

    set.add(morze)
  })

  return set.size
}

console.log(uniqueMorseRepresentations(["a"]))

File: go-archive/unique-morse-code-words.go
func uniqueMorseRepresentations(words []string) int {
    arr:=[]string{
        ".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}

    obj:=map[string]bool{}
    for i:=0; i < len(words); i++ {
        line:=""
        for j:=0; j < len(words[i]); j++{
            line+=arr[int(words[i][j])-97]
        }

        if (obj[line]) {
            continue
        } else {
            obj[line] = true
        }
    }

    return len(obj)
}